src/main/java/com/amazon/opendistroforelasticsearch/security/compliance/ComplianceConfig.java

```diff
@@ -41,209 +56,302 @@
+        final boolean logInternalConfig = settings.getAsBoolean(ConfigConstants.OPENDISTRO_SECURITY_COMPLIANCE_HISTORY_INTERNAL_CONFIG_ENABLED, false);
+        final boolean logReadMetadataOnly = settings.getAsBoolean(ConfigConstants.OPENDISTRO_SECURITY_COMPLIANCE_HISTORY_READ_METADATA_ONLY, false);
+        final boolean logWriteMetadataOnly = settings.getAsBoolean(ConfigConstants.OPENDISTRO_SECURITY_COMPLIANCE_HISTORY_WRITE_METADATA_ONLY, false);
+        final boolean logDiffsForWrite = settings.getAsBoolean(ConfigConstants.OPENDISTRO_SECURITY_COMPLIANCE_HISTORY_WRITE_LOG_DIFFS, false);
+        final List<String> watchedReadFields = settings.getAsList(ConfigConstants.OPENDISTRO_SECURITY_COMPLIANCE_HISTORY_READ_WATCHED_FIELDS,
+                Collections.emptyList(), false);
+        final List<String> watchedWriteIndices = settings.getAsList(ConfigConstants.OPENDISTRO_SECURITY_COMPLIANCE_HISTORY_WRITE_WATCHED_INDICES, Collections.emptyList());
+        final Set<String> immutableIndicesPatterns = ImmutableSet.copyOf(settings.getAsList(ConfigConstants.OPENDISTRO_SECURITY_COMPLIANCE_IMMUTABLE_INDICES, Collections.emptyList()));
+        final String saltAsString = settings.get(ConfigConstants.OPENDISTRO_SECURITY_COMPLIANCE_SALT, ConfigConstants.OPENDISTRO_SECURITY_COMPLIANCE_SALT_DEFAULT);
+        final String opendistrosecurityIndex = settings.get(ConfigConstants.OPENDISTRO_SECURITY_CONFIG_INDEX_NAME, ConfigConstants.OPENDISTRO_SECURITY_DEFAULT_CONFIG_INDEX);
+        final String type = settings.get(ConfigConstants.OPENDISTRO_SECURITY_AUDIT_TYPE_DEFAULT, null);
+        final String index = settings.get(ConfigConstants.OPENDISTRO_SECURITY_AUDIT_CONFIG_DEFAULT_PREFIX + ConfigConstants.OPENDISTRO_SECURITY_AUDIT_ES_INDEX, "'security-auditlog-'YYYY.MM.dd");
+
+        return new ComplianceConfig(
+                logExternalConfig,
+                logInternalConfig,
+                logReadMetadataOnly,
+                logWriteMetadataOnly,
+                logDiffsForWrite,
+                watchedReadFields,
+                watchedWriteIndices,
+                immutableIndicesPatterns,
+                saltAsString,
+                opendistrosecurityIndex,
+                type,
+                index);
+    }
 
-	public void setExternalConfigLogged(boolean externalConfigLogged) {
-		this.externalConfigLogged = externalConfigLogged;
-	}
+    /**
+     * Checks if config defined in elasticsearch config directory must be logged
+     * @return true/false
+     */
+    public boolean shouldLogExternalConfig() {
+        return logExternalConfig;
+    }
+
+    /**
+     * Checks if internal config must be logged
+     * @return true/false
+     */
+    public boolean shouldLogInternalConfig() {
+        return logInternalConfig;
+    }
 
-	public boolean isEnabled() {
-        return this.enabled;
+    /**
+     * Checks if compliance is enabled
+     * @return true/false
+     */
+    public boolean isEnabled() {
+        return true;
```

Comment:
I see you have changed values with the class member variables. Why opposite here?

Reply:
I will revert this back to class member variable. 

I actually wanted to remove this isEnabled() itself because its always set to true. I believe this is something we inherited from search-guard where compliance was offered as a separate feature that could be disabled. 
Other places check for this explicitly hence I retained for some future use-cases.